tracee-ebpf/tracee/tracee.go

```diff
@@ -177,37 +178,22 @@ type Tracee struct {
 	processTree       *ProcessTree
 }
 
-type counter int32
-
-func (c *counter) Increment(amount ...int) {
-	sum := 1
-	if len(amount) > 0 {
-		sum = 0
-		for _, a := range amount {
-			sum = sum + a
-		}
-	}
-	atomic.AddInt32((*int32)(c), int32(sum))
-}
-
 type statsStore struct {
-	eventCounter  counter
-	errorCounter  counter
-	lostEvCounter counter
-	lostWrCounter counter
-	lostNtCounter counter
+	eventCounter  counter.Counter
+	errorCounter  counter.Counter
+	lostEvCounter counter.Counter
+	lostWrCounter counter.Counter
+	lostNtCounter counter.Counter
 }
 
-func (t *Tracee) GetStats() external.Stats {
-	var stats external.Stats
-
-	stats.EventCount = int(t.stats.eventCounter)
-	stats.ErrorCount = int(t.stats.errorCounter)
-	stats.LostEvCount = int(t.stats.lostEvCounter)
-	stats.LostWrCount = int(t.stats.lostWrCounter)
-	stats.LostNtCount = int(t.stats.lostNtCounter)
-
-	return stats
+func (t *Tracee) GetStats() stats.EbpfStats {
+	return stats.EbpfStats{
+		EventCount:  int(t.stats.eventCounter),
```

Comment:
Read should be done atomically in order to prevent race condition with`Increment`-ing logics that can happen concurrently

Reply:
Done through counter.Read()